src/battle.ts

```diff
@@ -245,26 +245,26 @@ class Pokemon {
 		return this.ident.substr(0, 2) + slots[this.slot] + this.ident.substr(2);
 	}
 	removeVolatile(volatile: ID) {
-		this.sprite.removeEffect(volatile);
+		this.side.battle.scene.removeEffect(this, volatile);
```

Comment:
`this.battle` exists on every `Pokemon` instance; there's no need to go through `.side`